### PR TITLE
Bug 1183809 – Localize suggested sites

### PR DESCRIPTION
**Pull request by jhugman**
* Renamed SuggestedSitesData -> SuggestSitesCursor
* Added struct SuggestedSiteData, which was too close to its plural.
* Replace suggestedsites.json with DefaultSuggestedSites.sites, a struct array.
* Localized title and url.

### CODE REVIEW

File: Storage/DefaultSuggestedSites.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+public class DefaultSuggestedSites {
+    public static let sites = [
+        SuggestedSiteData(
+            url: NSLocalizedString("https://www.mozilla.org/about", comment: "Tile url for Mozilla"),
+            bgcolor: "0xce4e41",
+            imageurl: "asset://suggestedsites_mozilla",
```
Comment (sleroux) [inline Storage/DefaultSuggestedSites.swift]: Nit: We should be consistent with our camel casing on these properties: bgcolor -> bgColor, imageurl -> imageUrl

File: Storage/DefaultSuggestedSites.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+public class DefaultSuggestedSites {
+    public static let sites = [
+        SuggestedSiteData(
+            url: NSLocalizedString("https://www.mozilla.org/about", comment: "Tile url for Mozilla"),
```
Comment (st3fan) [inline Storage/DefaultSuggestedSites.swift]: Dont' localize the URLs. Just the titles. The URL will auto-redirect to the correct locale. There is nothing to localize for the L10N team there.

File: Storage/DefaultSuggestedSites.swift (hunk 1 above)
Comment (jhugman) [inline Storage/DefaultSuggestedSites.swift]: @sleroux 
I was torn between keeping consistent with the JSON file for android.json and beautiful camel case. :)

**Comment by jhugman** [comment]
Addressed nits and squashed.

**Pull request merged**